vars/washeryCleanSnapshots.groovy

```diff
@@ -0,0 +1,105 @@
+/***********************************
+washeryCleanSnapshots DSL
+
+Cleans old washery snapshots from RDS Snapshot storage
+
+example usage
+executeChangeSet(
+  region: 'us-east-1', // (required, aws region to deploy the stack)
+  accountId: env.DEV_ACCOUNT_ID,
+  role: env.CIINABOXV2_ROLE, // IAM role to assume
+  snapshotRetainCount: 3 // The number of washery snapshots to keep stored
+)
+************************************/
+
+import com.base2.ciinabox.aws.AwsClientBuilder
+import com.amazonaws.services.rds.model.DescribeDBSnapshotsRequest
+import com.amazonaws.services.rds.model.DeleteDBSnapshotRequest
+import com.amazonaws.services.rds.model.DescribeDBClusterSnapshotsRequest
+import com.amazonaws.services.rds.model.DeleteDBClusterSnapshotRequest
+
+def call(body) {
+    def config = body
+    def clientBuilder = new AwsClientBuilder([
+        region: config.region,
+        awsAccountId: config.accountId,
+        role: config.role
+    ])  
+
+    def client = clientBuilder.rds()
+
+    cleanInstanceWasherySnapshots(client, config.snapshotRetainCount)
+    cleanClusterWasherySnapshots(client, config.snapshotRetainCount)
+}
+
+@NonCPS
+def cleanInstanceWasherySnapshots(client, snapshotRetainCount){
+    //RDS Instance
+    def request = new DescribeDBSnapshotsRequest()
+            .withSnapshotType("manual")
+    def snapshotsResult = client.describeDBSnapshots(request)
+    def snapshots = snapshotsResult.getDBSnapshots()
+    def washerySnapshots = []
+
+    //Retrieve snapshots prefixed with `washery-scrubbed`
+    if(snapshots.size() > 0) {
+        for (int i = 0; i < snapshots.size(); i++) {
+            if (snapshots[i].getDBSnapshotIdentifier().startsWith("washery-scrubbed-")){
+                washerySnapshots.add(snapshots[i])
+            }
+        }
+        //Sort washery snapshots based on snapshot create time
+        washerySnapshots.sort{a,b-> a.getSnapshotCreateTime().compareTo(b.getSnapshotCreateTime())}
+        for (int i = 0; i < washerySnapshots.size(); i++) {
+            echo "${washerySnapshots[i].getDBSnapshotIdentifier()} - ${washerySnapshots[i].getSnapshotCreateTime().format('d/M/yyyy HH:mm:ss')}"
+        }
+    }
+
+    //Delete snapshot's until only the snapshotRetainCount amount remains
+    while (washerySnapshots.size() > snapshotRetainCount){
```

Comment:
if you split the snapshot list into keep and delete you could for loop over the delete list

```groovy
for (snapshot in washerySnapshots) {
  snapshot_identifier = snapshot.getDBSnapshotIdentifier()
  def delete_request = new DeleteDBSnapshotRequest().withDBSnapshotIdentifier(snapshot_identifier)
  def response = client.deleteDBSnapshot(delete_request)
  echo "Deleted Snapshot - ${snapshot_identifier} created on ${current_snapshot.getSnapshotCreateTime()}"
}

Reply:
Added cleaner delete loop
```
 //Delete snapshot's until only the snapshotRetainCount amount remains
        for (snapshot in delete) {
            snapshot_identifier = snapshot.getDBSnapshotIdentifier()
            def delete_request = new DeleteDBSnapshotRequest().withDBSnapshotIdentifier(snapshot_identifier)
            def response = client.deleteDBSnapshot(delete_request)
            echo "Deleted Snapshot - ${snapshot_identifier} created on ${snapshot.getSnapshotCreateTime()}"
        }
```